test/e2e/deployment.go

```diff
@@ -644,8 +634,9 @@ func testRollbackDeployment(f *Framework) {
 	err = waitForDeploymentStatus(c, ns, deploymentName, deploymentReplicas, deploymentReplicas-1, deploymentReplicas+1, 0)
 	Expect(err).NotTo(HaveOccurred())
 
-	// Check if it's updated to revision 2 correctly
-	checkDeploymentRevision(c, ns, deploymentName, "2", updatedDeploymentImageName, updatedDeploymentImage)
+	// Wait for it to be updated to revision 2
```

Comment:
I think this should this go before waitForDeploymentStatus, to ensure the status is about the image we care about.